pipeline/src/node_properties/function_properties.rs

```diff
@@ -2,5 +2,20 @@ use serde::{Deserialize, Serialize};
 
 #[derive(Clone, Debug, Deserialize, PartialEq, Serialize)]
 pub struct FunctionProperties {
-    pub module: String,
+    pub code: String,
+    #[serde(flatten)]
+    pub runtime: FunctionRuntime,
+}
+
+#[derive(Clone, Debug, Deserialize, PartialEq, Serialize)]
+pub struct FunctionRuntime {
+    /// Path to python module
+    ///
+    /// This field is set to `Some` by `lumeod`
+    ///
+    /// `lumeod` receives code as a string blob (`FunctionProperties::code`) but gvapython
+    /// element requires a path to ".py" file, so lumeod stores code in a file and sets
+    /// `module` field to file path.
+    // TODO: replace String with fs::Path
```

Comment:
```suggestion
    // TODO: replace String with fs::PathBuf
```

`PathBuf` is the filesystem path equivalent to `String`, `Path` is the equivalent to `str`.